lib/online_kitchen/labmanager4.rb

```diff
@@ -0,0 +1,165 @@
+# = Wrapper for Online Kitchen SOAP service
+#
+# This library is an interface for Online Kitchen service which is an interface
+# for creating and destroing virtual machines (currently only VMware vsphere)
+#
+# == Example
+#
+# lm = OnlineKitchen::LabManager.create(
+#   vms_folder: 'myFolder',
+#   template_name: 'w7x64',
+#   requestor: 'CZ\\user.name',
+#   uuid: ...,
+#   job_id: ...
+# )
+# p lm.ip
+# lm.destroy
+# ...
+#
+# # destroy machine by name
+# OnlineKitchen::LabManager.destroy('online_kitchen_8x64-5e448e10')
+#
+
+require 'securerandom'
+require 'net/http'
+require 'json'
+require 'active_support/core_ext/object/blank'
+
+module OnlineKitchen
+  # LabManager communication encapsulation
+  class LabManager4
+    attr_reader :vm
+
+    class << self
+      def create(opts = {})
+        new.provision_machine(opts)
+      end
+
+      def release_machine(name)
+
+          config = OnlineKitchen.config.rest_config
+          machine_id = /.*\-([^\-]*)/.match(name)[1]
+          uri_del_machine = URI("#{config[:service_endpoint]}machines/#{machine_id}")
+
+          request_id = Net::HTTP.start(uri_del_machine.host, uri_del_machine.port) do |http|
+            request = Net::HTTP::Delete.new uri_del_machine
+            request.basic_auth config[:username], config[:password]
+
+            response = http.request request # Net::HTTPResponse object
+            OnlineKitchen.logger.debug(response.body)
+            JSON.parse(response.body)["request_id"]
+          end
+
+          OnlineKitchen.logger.debug(request_id)
+      end
+
+      alias destroy release_machine
+
+    end
+
+    def initialize
+      @vm = {}
+    end
+
+    def destroy
+      self.class.release_machine(vm[:name]) if vm && vm[:name]
+      @vm = {}
+      self
+    end
+
+    def ip
+      vm[:ip]
+    end
+
+    def name
+      vm[:name]
+    end
+
+    def wait_for_machine_deployed(config, machine_id)
+        rand_gen = Random.new
+        for idx in 1..40 do
```

Comment:
It would be nice to have these magic constants in some configuration file.

Reply:
Great notice, fix it youself immediatelly :)